File: web/src/types/supplier.ts
/* ========================================
   供应商模块类型（对齐 OpenAPI 契约）
   ======================================== */

/** 供应商详情（对齐 Supplier schema） */
export interface Supplier {
  id: number;
  code: string;
  name: string;
  address: string;
  contact?: string | null;
  phone?: string | null;
  email?: string | null;
  remark?: string | null;
  status: 0 | 1;
  createdAt: string;
  updatedAt: string;
  createdBy?: number | null;
  updatedBy?: number | null;
}

/** 供应商列表项（对齐 SupplierSummary schema） */
export interface SupplierSummary {
  id: number;
  code: string;
  name: string;
  address: string;
  contact?: string | null;
  phone?: string | null;
  status: 0 | 1;
  createdAt: string;
}

/** 供应商列表分页响应 */
export interface SupplierListResponse {
  items: SupplierSummary[];
  total: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

/** 创建供应商请求 */
export interface CreateSupplierRequest {
  code: string;
  name: string;
  address: string;
  contact?: string;
  phone?: string;
  email?: string;
  remark?: string;
  status?: 0 | 1;
}

/** 更新供应商请求（code 不可改，状态走独立接口） */
export interface UpdateSupplierRequest {
  name?: string;
  address?: string;
  contact?: string;
  phone?: string;
  email?: string;
  remark?: string;
}

/** 供应商查询参数 */
export interface SupplierQueryParams {
  keyword?: string;
  status?: 0 | 1;
  page?: number;
  pageSize?: number;
}

File: api/src/main/java/com/aistore/module/supplier/entity/Supplier.java
package com.aistore.module.supplier.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/** 供应商实体（表 supplier） */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("supplier")
public class Supplier {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;
    private String code;
    private String name;
    private String address;
    private String contact;
    private String phone;
    private String email;
    private String remark;
    private Integer status;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    private Long createdBy;
    private Long updatedBy;

    @TableLogic
    private Integer deleted;
}
